README.md

```diff
@@ -117,3 +159,55 @@ See [LICENSE](https://github.com/monte-carlo-data/apollo-agent/blob/main/LICENSE
 ## Security
 
 See [SECURITY](https://github.com/monte-carlo-data/apollo-agent/blob/main/SECURITY.md) for more information.
+
+## Advanced deployment
+
+### Cloud Run deployment
+- You need to have `gcloud` CLI configured, instructions [here](https://cloud.google.com/sdk/docs/install-sdk).
+- Once your gcloud cli is configured you can deploy the application using:
+```shell
+gcloud run deploy CLOUD_RUN_SERVICE_NAME_HERE --image montecarlodata/pre-release-agent:latest-cloudrun
+```
+- The previous step assumes the code to deploy is the latest in the `dev` branch, you can also deploy 
+  your local code (passing `--source .` instead of `--image`), but you also need to update the `Dockerfile` 
+  to leave only the `cloudrun` image as there's no way to specify the target and it seems the last one
+  is automatically selected.
+- If you enable authentication, you'll need to create a service account to use as the "invoker", the minimum 
+  required permission for that service account is "Cloud Run Invoker" for the deployed service
+- Once you configured the service account to use as the client, you can create identity tokens using:
+```shell
+gcloud auth print-identity-token [service account email address]
+```
+- The identity token is sent as the Bearer authorization header: `Authorization: Bearer IDENTITY_TOKEN`
+
+To update an agent in CloudRun you need to run the same command used to deploy it for the first time:
+```shell
+gcloud run deploy CLOUD_RUN_SERVICE_NAME_HERE --image montecarlodata/pre-release-agent:latest-cloudrun
+```
+
+### Azure deployment
+You can check the README file for Azure [here](apollo/interfaces/azure/README.md).
+For authentication you need to get the app-key for the Azure App and pass it in the `x-functions-key` header.
+
+### Lambda deployment
+You can build the Docker image for the Lambda agent using:
+```shell
+docker build -t lambda_agent -f Dockerfile --target lambda --build-arg code_version=0.0.2 --build-arg build_number=106 --platform=linux/amd64 .
+```
+
+If you have your own ECR repo used for testing you'll need to login first:
+```shell
+aws ecr get-login-password --region us-east-1 --profile <aws_profile> | docker login --username AWS --password-stdin <account_id>.dkr.ecr.us-east-1.amazonaws.com
+```
+
+And then tag/push your image:
+```shell
+docker tag lambda_agent:latest <account_id>.dkr.ecr.us-east-1.amazonaws.com/dev-agent:95
+docker push <account_id>.dkr.ecr.us-east-1.amazonaws.com/dev-agent:95 
```

Comment:
These steps hardcode version numbers. We should make it clear that those numbers are just examples